src/adc121c021/adc121c021.cxx

```diff
@@ -79,12 +79,20 @@ mraa_result_t ADC121C021::writeWord(uint8_t reg, uint16_t word)
 
 uint8_t ADC121C021::readByte(uint8_t reg)
 {
-  return mraa_i2c_read_byte_data(m_i2c, reg);
+  int val = mraa_i2c_read_byte_data(m_i2c, reg);
+  if (val != -1) {
+    return (uint8_t) val;
+  }
+  return 0;
```

Comment:
This might be problematic, since a failure might look like a valid reading. For example, this chip's "Alert Status Register" and "Configuration Register" both have a default value `0`, so reading either and getting the default value has the same outcome as reading reading it but the read fails for whatever reason that would cause this condition: https://github.com/intel-iot-devkit/mraa/blob/6d8ec87e0e0561b27c1c57afcf47fbbd14aec3f3/src/i2c/i2c.c#L233-L244


`mraa`'s cpp API throws an exception: https://github.com/intel-iot-devkit/mraa/blob/c1465bd69463b34c477eea503f14716c70ed48d0/api/mraa/i2c.hpp#L140-L148 which is probably better?